operators/bosh-restore-operator/BoshRestoreService.js

```diff
@@ -0,0 +1,548 @@
+const catalog = require('../../common/models/catalog');
+const backupStore = require('../../data-access-layer/iaas').backupStore;
+
+class BoshRestoreService extends BaseDirectorService {
+  constructor(plan) {
+    super(plan);
+    this.plan = plan;
+    this.cloudProvider = cloudProvider;
+    this.director = bosh.director;
+    this.backupStore = backupStore;
+  }
+
+  async startRestore(opts) { 
+    try {
+      logger.debug('Starting restore with options:', opts);
+      const args = opts.arguments;
+      const backupMetadata = _.get(args, 'backup');
+      const deploymentName = await this.findDeploymentNameByInstanceId(opts.instance_guid); 
+
+      const data = _
+        .chain(opts)
+        .pick('service_id', 'plan_id', 'instance_guid', 'username')
+        .assign({
+          operation: 'restore',
+          backup_guid: args.backup_guid,
+          time_stamp: args.time_stamp,
+          state: 'processing',
+          started_at: new Date().toISOString(),
+          finished_at: null,
+          tenant_id: opts.context ? this.getTenantGuid(opts.context) : args.space_guid
+        })
+        .value();
+
+      const service = catalog.getService(opts.service_id);
+      const instanceGroups = _.get(service, 'restore_operation.instance_group');
+      let persistentDiskInfo = await this.director.getPersistentDisks(deploymentName, instanceGroups); 
+
+      let getDiskMetadataFn = async instance => {
+        let diskCid = instance.disk_cid;
+        let az = instance.az;
+        instance.oldDiskInfo = await this.cloudProvider.getDiskMetadata(diskCid, az);
+      };
+
+      await Promise.all(persistentDiskInfo.map(getDiskMetadataFn)); 
+
+      const optionsData = _
+        .assign({
+          restoreMetadata: {
+            timeStamp: args.time_stamp,
+            filePath: _.get(service, 'restore_operation.filesystem_path'),
+            snapshotId: _.get(backupMetadata, 'snapshotId'),
+            deploymentName: deploymentName,
+            deploymentInstancesInfo: persistentDiskInfo,
+            baseBackupErrand: {
+              name: _.get(service, 'restore_operation.errands.base_backup_restore.name'),
+              instances: _.get(service, 'restore_operation.errands.base_backup_restore.instances')
+            },
+            pointInTimeErrand: {
+              name: _.get(service, 'restore_operation.errands.point_in_time.name'),
+              instances: _.get(service, 'restore_operation.errands.point_in_time.instances')
+            },
+            postStartErrand: {
+              name: _.get(service, 'restore_operation.errands.post_start.name'),
+              instances: _.get(service, 'restore_operation.errands.post_start.instances')
+            }
+          },
+          stateResults: {}
+        });
+
+      let restoreFileMetadata;
+      try {
+        restoreFileMetadata = await this.backupStore.getRestoreFile(data);
+      } catch(err) {
+        if (!(err instanceof errors.NotFound)) {
```

Comment:
You can directly put `errors.NotFound` in catch and throw.

Reply:
Actually expected behaviour is that to throw for all errors _except_ `errors.NotFound`, hence explicit check.